test/math/SafeMath.test.js

```diff
@@ -10,134 +10,264 @@ contract('SafeMath', function () {
         const a = new BigNumber(5678);
-        const b = new BigNumber(5678);
+        const b = new BigNumber(1234);
 
-        (await this.safeMath.mod(a, b)).should.be.bignumber.equal(a.mod(b));
+        const result = await this.safeMath.subInts(a, b);
+        result.should.be.bignumber.equal(a.minus(b));
       });
 
-      it('when the dividend is larger than the divisor', async function () {
-        const a = new BigNumber(7000);
+      it('subtracts correctly if it does not overflow and the result is negative', async function () {
+        const a = new BigNumber(1234);
         const b = new BigNumber(5678);
 
-        (await this.safeMath.mod(a, b)).should.be.bignumber.equal(a.mod(b));
+        const result = await this.safeMath.subInts(a, b);
+        result.should.be.bignumber.equal(a.minus(b));
       });
 
-      it('when the dividend is a multiple of the divisor', async function () {
-        const a = new BigNumber(17034); // 17034 == 5678 * 3
+      it('reverts on positive subtraction overflow', async function () {
+        const a = MAX_INT256;
+        const b = new BigNumber(-1);
+
+        await shouldFail.reverting(this.safeMath.subInts(a, b));
+      });
+
+      it('reverts on negative subtraction overflow', async function () {
+        const a = MIN_INT256;
+        const b = new BigNumber(1);
+
+        await shouldFail.reverting(this.safeMath.subInts(a, b));
+      });
+    });
+
+    describe('mul', function () {
+      it('multiplies correctly', async function () {
+        const a = new BigNumber(5678);
+        const b = new BigNumber(-1234);
+
+        const result = await this.safeMath.mulInts(a, b);
+        result.should.be.bignumber.equal(a.times(b));
+      });
+
+      it('handles a zero product correctly', async function () {
+        const a = new BigNumber(0);
         const b = new BigNumber(5678);
 
-        (await this.safeMath.mod(a, b)).should.be.bignumber.equal(a.mod(b));
+        const result = await this.safeMath.mulInts(a, b);
+        result.should.be.bignumber.equal(a.times(b));
+      });
+
+      it('reverts on multiplication overflow, positive operands', async function () {
+        const a = MAX_INT256;
+        const b = new BigNumber(2);
+
+        await shouldFail.reverting(this.safeMath.mulInts(a, b));
+      });
+
+      it('reverts when minimum integer is multiplied by -1', async function () {
+        const a = MIN_INT256;
+        const b = new BigNumber(-1);
+
+        await shouldFail.reverting(this.safeMath.mulInts(a, b));
+      });
+
+      it('reverts when -1 is multiplied by minimum integer', async function () {
+        const a = new BigNumber(-1);
+        const b = MIN_INT256;
+
+        await shouldFail.reverting(this.safeMath.mulInts(a, b));
       });
     });
 
-    it('reverts with a 0 divisor', async function () {
-      const a = new BigNumber(5678);
-      const b = new BigNumber(0);
+    describe('div', function () {
```

Comment:
Unsigned integer division has a couple more tests this one is missing, should we add them?